packages/core/src/readers/PDFReader.ts

```diff
@@ -1,33 +1,35 @@
+import { fs } from "@llamaindex/env";
 import { Document } from "../Node.js";
 import { FileReader } from "./type.js";
 
 /**
  * Read the text of a PDF
  */
 export class PDFReader extends FileReader {
-  async loadDataAsContent(fileContent: Buffer): Promise<Document[]> {
-    const pages = await readPDF(fileContent);
-    return pages.map((text, page) => {
+  async loadData(file: string): Promise<Document[]> {
```

Comment:
@himself65 don't override `loadData` from `FileReader` - that way you're losing the unified metadata. Better make `loadData` `final` in `FileReader`?

Also `FileReader` is using `loadDataAsContent(content: Buffer)` instead of `loadDataAsContent(content: Uint8Array)` - how do you suggest dealing with this?